### ISSUE TITLE
FR: Homescreen widget for Android

### ISSUE THREAD

**Issue by uroybd** [koreader/koreader#12626]
There are many ereader app in Android providing homescreen widget for latest reading. Unfortunately, none of those is KOReader, and I cannot use anything but KOReader now.

I know it is a complicated and one-platform-only feature but can it be done?

I wish I could contribute directly to this feature, however, my knowledge in android development is poor.

**Comment by Frenzie** [comment]
I think this may have been asked before. A widget as such looks like it should be relatively simple, but what it should do and getting data to it much less so.
https://developer.android.com/develop/ui/views/appwidgets

**Comment by pazos** [comment]
> I know it is a complicated and one-platform-only feature but can it be done?

It can be done but we don't want to :).

Why?

Because it has nothing to do with KOReader development, provides no benefit for the program and it is a bunch of work.

Like #9134, I think it is out of the scope. Please implement elsewhere. 

We can make a public service, available to the apps that request a given permission, that returns chunks of metadata. Maybe we can send thumbnails too using base64 encoded strings?.

I don't know. If I were to create something from scratch I would go for #9134, as they look mostly the same amount of work and it is better to avoid KOReader as a source of trusted metadata :)

Or the same app could contain both a home activity and widgets...

tl;dr: we can provide an API, never implement the full thing here.

**Comment by uroybd** [comment]
> We can make a public service, available to the apps that request a given permission, that returns chunks of metadata. Maybe we can send thumbnails too using base64 encoded strings?

With this, I can manage to make a launcher/widget separately.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
